src/Processors/Transforms/DistinctSortedChunkTransform.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <Columns/IColumn.h>
+#include <Core/ColumnNumbers.h>
+#include <Core/SortDescription.h>
+#include <Interpreters/SetVariants.h>
+#include <Processors/ISimpleTransform.h>
+
+namespace DB
+{
+
+///
+/// DISTINCT optimization for sorted chunks
+/// It also checks if current chunks is continuation of previous one,
+/// i.e. sorting prefix value of last row in previous chunk is the same as of first row in current one,
+/// so it can correctly process sorted stream as well
```

Comment:
it worth to mention here the logic depending on `other_columns.empty()`. if true, we will go without hash tables at all. if false - HT will be created for each value of sorting key prefix which means lower memory consumption.